service/geopmdpy/service.py

```diff
@@ -514,6 +514,10 @@ def start_batch(self, client_pid, signal_config, control_config):
         elif not cont_req.issubset(supported_controls):
             raise RuntimeError('Requested controls that are not in allowed list: {}' \
                                .format(sorted(cont_req.difference(supported_controls))))
+        for domain, domain_idx in [(cc[0], cc[1]) for cc in signal_config]:
+            self._check_domain_permissions(client_pid, domain, domain_idx)
+        for domain, domain_idx in [(cc[0], cc[1]) for cc in control_config]:
```

Comment:
I would refactor these loops. They look somewhat hard to read and understand, in such a compact format. Looks like a very cool code trick, but somewhat obfuscated.

Reply:
```
        for domain, domain_idx in [(cc[0], cc[1]) for cc in signal_config]:
            self._check_domain_permissions(client_pid, domain, domain_idx)
```

The other less readable way to write this is:

```
        for cc in signal_config:
            self._check_domain_permissions(client_pid, cc[0], cc[1])
```

I'm not sure which is less confusing for the reader, but the original implementation is just to enable the values to be named in the function call.